documentation/specs/event-source.md

```diff
@@ -0,0 +1,25 @@
+# EventSource
+
+[EventSource](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource?view=netframework-4.8) is the tool that allows Event Tracing for Windows (ETW) used in MSBuild. Among its useful features, functions with names ending in "start" and "stop" correlate between calls such that it can automatically record how long the event between the two calls took. It also provides an easy way to cheaply opt in or out, log auxiliary messages in addition to time, and add progress updates in the middle of an event as needed.
```

Comment:
Do the integers in the EventSource correlation ID map exactly to the various MSBuild IDs? If not, can they be made to?

Reply:
What do you mean by correlation ID and MSBuild IDs?

Reply:
the "EventSource correlation ID" I mentioned is displayed as the ActivityID in PerfView:

![image](https://user-images.githubusercontent.com/3347530/66141716-c5216c80-e5c9-11e9-8e2d-19762e87ee25.png)

My question is: do the various integers there match up with MSBuild's internal IDs like `ProjectID` or `BuildRequest.GlobalRequestId`? If not, can they be made to do so?

Reply:
I believe MSBuild's internal IDs are assigned by MSBuild at build time, whereas these numbers are assigned by a combination of the eventsource code and how many times a particular event has run within this build, so they cannot be made to align, at least not from this side. It's feasible that MSBuild could choose numbers based on how eventsource will work.